Testscripts/Windows/VMRESIZE-HB.ps1

```diff
@@ -0,0 +1,254 @@
+		} else {
+			Write-LogErr "$($vmStatus.Statuses[1].DisplayStatus): Could not find the VM status before hibernation"
+			throw "Can not identify VM status before hibernate"
+		}
+
+		# Hibernate the VM
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./test.sh" -runAsSudo -RunInBackground -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Sent hibernate command to the VM and continue checking its status in every 15 seconds until 10 minutes timeout "
+
+		# Verify the VM status
+		# Can not find if VM hibernation completion or not as soon as it disconnects the network. Assume it is in timeout.
+		$timeout = New-Timespan -Minutes 10
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout){
+			Wait-Time -seconds 15
+			$vmStatus = Get-AzVM -Name $vmName -ResourceGroupName $rgName -Status
+			if ($vmStatus.Statuses[1].DisplayStatus -eq "VM stopped") {
+				break
+			} else {
+				Write-LogInfo "VM status is not stopped. Waiting for 15s..."
+			}
+		}
+		if ($vmStatus.Statuses[1].DisplayStatus -eq "VM stopped") {
+			Write-LogInfo "$($vmStatus.Statuses[1].DisplayStatus): Verified successfully VM status stopped after hibernation command sent"
+		} else {
+			Write-LogErr "$($vmStatus.Statuses[1].DisplayStatus): Could not verify the VM status stopped after hibernation command sent"
+			throw "Did not verify VM status stopped after hibernate"
+		}
+
+		# Resize Azure VM
+		Write-LogInfo "Verified the current VM size is $($vm.HardwareProfile.VmSize) and change to new size Standard_D4s_v3"
+		$vm.HardwareProfile.VmSize = "Standard_D4s_v3"
+		Write-LogInfo "Updating VM size ..."
+		$updated_vm = Update-AzVM -VM $vm -ResourceGroupName $rgName
+		if (($updated_vm.StatusCode -eq 'OK') -and $updated_vm.IsSuccessStatusCode) {
+			Write-LogInfo "Successfully changed the VM size"
+		} else {
+			$vm = Get-AzVM -Name $vmName -ResourceGroupName $rgName
+			throw "Failed to change the VM size. Expected Standard_D4s_v3, but $($vm.HardwareProfile.VmSize). $($updated_vm.ReasonPhase)"
+		}
+
+		Write-LogInfo "VM $vmName in Resource Group $rgName is restarting becasue of different memory size or VM size. Continue checking its status in every 15 seconds until 15 minutes timeout "
+
+		# Wait for VM starting for 15 min-timeout
+		$timeout = New-Timespan -Minutes 15
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout) {
+			Wait-Time -seconds 15
+			$state = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password -command "date > /dev/null; echo $?"
+			if ($state) {
+				Write-LogInfo "VM is now accessible remotely"
+				break
+			} else {
+				Write-LogInfo "VM is still resuming!"
+			}
+		}
+
+		if ($state){
+			Write-LogInfo "VM restart completed"
+		} else {
+			throw "VM start halt or hang, the latest state of remote connection was $state"
```

Comment:
at this moment $state is $null or empty

Reply:
Fixed in 7e8639e54beeb09f5ac14bc0f1df80a6727a60b2